docs/configuration.md

```diff
@@ -115,6 +115,8 @@ of the most common options to set are:
   <td>
     Amount of memory to use for the driver process, i.e. where SparkContext is initialized.
     (e.g. <code>512m</code>, <code>2g</code>).
+    
+    <br /><br />Note: this setting only works in <code>cluster</code> mode (e.g. YARN deployment). In <code>client</code> mode (e.g. spark-shell), this setting has no effect. In client mode, driver memory should be configured in the run-time settings ; i.e. --driver-memory 2g.
```

Comment:
By the way this explanation isn't exactly true. It only doesn't work if you set them through the `conf.set(...)` way. It still works if you put them in your `conf/spark-defaults.conf` file.